src/app/pages-block/services/courses.service.ts

```diff
@@ -1,36 +1,43 @@
+import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
 import { CoursePage } from 'src/app/interfaces/classes';
 
 @Injectable({
   providedIn: 'root'
 })
 export class CoursesService {
 
-  private courses: CoursePage[] = [
-    new CoursePage('1', 'First', new Date(2021, 1, 12), 121, 'Some description Lorem ipsum dolor sit amet dolor sit amet consectetur consectetur dolor sit amet consectetur, adipisicing elit. Asperiores, est! Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, itaque.'),
-    new CoursePage('2', 'Second', new Date(2025, 5, 30), 59, 'Not good course.'),
-    new CoursePage('3', 'Third', new Date(2022, 6, 4), 96, 'Some description   elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwer sit amet consectetur, adipisicing elit. lorem lowerwer Asperiores, est! Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, itaque.'),
-  ];
   course: CoursePage;
+  currentCourseId: BehaviorSubject<string> = new BehaviorSubject<string>('');
 
-  getCoursesList(): CoursePage[] {
-    return this.courses
+  constructor(private http: HttpClient) { }
+
+  getCoursesList(courseLimit: number): Observable<CoursePage[]> {
+    return this.http.get<CoursePage[]>(`http://localhost:3000/courses?_limit=${courseLimit}`);
```

Comment:
you can move "http://localhost:3000/" to variable
then when in future you will need change BE source link you will need change it just in one place
or even better create interceptor which will add this part to every request